src/sentry/snuba/metrics/fields/base.py

```diff
@@ -233,7 +366,8 @@ class DerivedMetricDefinition:
     unit: str
     op: Optional[str] = None
     result_type: Optional[MetricType] = None
-    snql: Optional[Function] = None
+    # TODO: better typing
+    snql: Optional[Callable[..., Function]] = None
```

Comment:
I had to fix this because `snql` is a python function that returns a snql function expression. python functions are `Callable`, `Function` is from snuba-sdk